### DISCUSSION TITLE
Simple python app that reproduce default indexing

### DISCUSSION THREAD

**Discussion by MisterAndry** [microsoft/graphrag#694] [Q&A] [answered]
I'm curious if there is an example of using GraphRag with Python Api that exactly replicates the default command: 
`python -m graphrag.index --root . `
?
Right [here](https://microsoft.github.io/graphrag/posts/index/overview/)  example with Python Api doesn't look like the default command, as there is some step with built-in verb, dataset added manually, no reading of data from input folder, etc.

**Comment by flowtrader2016** [comment]
I was looking for this too.

**Comment by natoverse** [comment, marked as answer]
You probably want to look at the [CLI code](https://github.com/microsoft/graphrag/blob/27c5468a8b6277b7eb2f6e06a701aad55fff50f7/graphrag/index/cli.py#L99) itself, which loads config and executes the indexer.

**Reply by MisterAndry** [reply]
looks like a good advice! 
Thanks!

Here is example for v0.3.3:

```
from graphrag.index.cli import index_cli


def run_indexing():
    index_cli(root_dir="ABSOLUTE_PATH_TO_THE_ROOT_FOLDER_WHICH_CONTAINS_INPUT_FOLDER",
              init=False,
              verbose=None,
              resume=False,
              update_index_id=False,
              memprofile=None,
              nocache=None,
              reporter=None,
              config_filepath=False,
              emit=False,
              dryrun=False,
              skip_validations=False)


run_indexing()
```
